psd-web/app/services/change_case_owner.rb

```diff
@@ -14,15 +14,23 @@ def call
     return if old_owner == owner
 
     ActiveRecord::Base.transaction do
-      if owner.is_a?(Team)
-        # Remove the new owner team as a collaborator if they were one.
-        investigation.edit_access_collaborations.where(editor: owner).delete_all
+      unless owner.team == investigation.owner_team_collaboration.collaborator
+        investigation.owner_team_collaboration.swap_to_edit_access!
       end
 
-      investigation.update!(owner: owner)
+      investigation.reload # force cached associations to be reloaded
+
+      old_collaborator = investigation
+                           .collaboration_accesses
+                           .where.not(type: ["Collaboration::Access::OwnerUser", "Collaboration::Access::OwnerTeam"])
+                           .find_by(collaborator: owner)
```

Comment:
```suggestion
                           .find_by(collaborator: owner.team)
```
I think you might need to ensure you're checking for a `Team`, as `owner` could be a `User`.